plugins/woocommerce/includes/admin/class-wc-admin-marketplace-promotions.php

```diff
@@ -0,0 +1,297 @@
+ *
+ * @package  WooCommerce\Admin
+ * @version  2.5.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * WC_Admin_Marketplace_Promotions class.
+ */
+class WC_Admin_Marketplace_Promotions {
+
+	const TRANSIENT_NAME = 'wc_addons_marketplace_promotions';
+	public static string $locale;
+
+	/**
+	 * On selected WooCommerce admin pages, fetch promotions from transient or API.
+	 * On pages in the main WooCommerce menu, show promotions if there are any.
+	 *
+	 * @return void
+	 */
+	public static function init_marketplace_promotions() {
+		if ( self::is_targeted_page_to_fetch_promotions() ) {
+			self::fetch_marketplace_promotions();
+		}
+
+		if ( self::is_targeted_page_to_show_bubble_promotions() ) {
+			self::$locale = ( self::$locale ?? get_user_locale() ) ?? 'en_US';
+			self::show_bubble_promotions();
+		}
+	}
+
+	/**
+	 * Check if the request is for one of the pages we will run fetch_marketplace_promotions
+	 * on: the WooCommerce Home and Extensions pages.
+	 *
+	 * @return bool
+	 */
+	public static function is_targeted_page_to_fetch_promotions(): bool {
+		if (
+			defined( 'DOING_AJAX' ) && DOING_AJAX
+			|| ! is_admin()
+			|| ! isset( $_GET['page'] )
+		) {
+			return false;
+		}
+
+		$targeted_page = 'wc-admin';
+		$targeted_path = '/extensions';
+
+		if ( $_GET['page'] === $targeted_page ) {
+			// WooCommerce home has page param but no path param
+			return ! isset( $_GET['path'] ) || $_GET['path'] === $targeted_path;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Check if the request is for one of the pages we will show menu item bubbles
+	 * on: the pages in the main WooCommerce menu, excluding Analytics and Marketing.
+	 *
+	 * We need to cover all the pages in the WooCommerce menu, because the Extensions item
+	 * is visible on any of these pages, and we want to show the bubble there.
+	 *
+	 * @return bool
+	 */
+	public static function is_targeted_page_to_show_bubble_promotions(): bool {
+		if (
+			defined( 'DOING_AJAX' ) && DOING_AJAX
+			|| ! is_admin()
+			|| ! isset( $_GET['page'] )
+		) {
+			return false;
+		}
+
+		// We also target wc-admin, but that has more variations
+		$targeted_pages = array(
+			'wc-orders',
+			'wc-reports',
+			'wc-settings',
+			'wc-status',
+		);
+
+		if ( in_array( $_GET['page'], $targeted_pages ) ) {
+			return true;
+		}
+
+		$targeted_wc_admin_paths = array(
+			'/customers',
+			'/extensions',
+		);
+
+		if ( 'wc-admin' === $_GET['page'] ) {
+			// WooCommerce home has page param but no path param
+			return ! isset( $_GET['path'] ) || in_array( $_GET['path'], $targeted_wc_admin_paths );
+		}
+
+		return false;
+	}
+
+	/**
+	 * Get promotions to show in the Woo in-app marketplace.
+	 * Only run on selected pages in the main WooCommerce menu in wp-admin.
+	 * Loads promotions in transient with one day life.
+	 *
+	 * @return void
+	 */
+	private static function fetch_marketplace_promotions() {
+		$url        = 'https://woocommerce.com/wp-json/wccom-extensions/3.0/promotions';
```

Comment:
Maybe we could use the new URL?
```suggestion
		$url        = 'https://woo.com/wp-json/wccom-extensions/3.0/promotions';
```

Reply:
I was being cautious, maybe too cautious – changed in b7148ec716bcfcfe328f87f1b4350f9f1900d059.